### PR TITLE
Added basic unittest for instances.py

### PR DESCRIPTION
**Pull request by kelseylam**
Made very basic, bare minimum unittests for instances.py to test the main method and adding CLI arguments. I still need to make more thorough tests in the future.

### CODE REVIEW

File: test/cli/instances_test.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        Application is instantiated and run() is called in main() for instances.py
+        """
+        app = MagicMock()
+        app_class = MagicMock(return_value=app)
+
+        with patch('aws_analysis_tools.cli.instances.Application', app_class):
+            main()
+
+        app_class.assert_called_once_with()
+        app.run.assert_called_once_with()
```
Comment (joestump) [inline test/cli/instances_test.py]: I believe you should be passing arguments to `_with()` tests to ensure it was called with the arguments expected. If you just care that they were each called once, you can use `assert_called_once()` instead (or `self.assertTrue(some_mock_object.called)`).

**Comment by joestump** [comment]
🎉 tests!

File: test/cli/instances_test.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        self.assertIn('group', app.args)
+        self.assertIn('exclude_group', app.args)
+        self.assertIn('name', app.args)
+        self.assertIn('exclude_name', app.args)
+        self.assertIn('type', app.args)
+        self.assertIn('exclude_type', app.args)
+        self.assertIn('zone', app.args)
+        self.assertIn('exclude_zone', app.args)
+        self.assertIn('state', app.args)
+        self.assertIn('exclude_state', app.args)
```
Comment (hanpeter) [inline test/cli/instances_test.py]: There is `--no-name` argument in the `app.args`. Please test the existence of that as well.

File: test/cli/instances_test.py (hunk 1 above)
Comment (hanpeter) [inline test/cli/instances_test.py]: If you only care about the methods are called, then let's leave a note to augment the test later to consider the parameters.

**Comment by hanpeter** [comment]
My comments are not show stoppers. :+1:

**Pull request merged**